src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -198,11 +201,19 @@ private static Result handleStatementTerminatorTyped(@NotNull Project project,
         if (formattingScope == FormattingScope.STATEMENT) {
             List<TextRange> selectionTextRanges = LSPSelectionRangeSupport.getSelectionTextRanges(file, editor, beforeOffset);
             if (!ContainerUtil.isEmpty(selectionTextRanges)) {
-                // Find the closest selection range that is extended to line start/end; that should be the statement
                 Document document = editor.getDocument();
                 CharSequence documentChars = document.getCharsSequence();
+
+                // Extend these to full lines for evaluation below
+                Set<TextRange> extendedSelectionTextRanges = new LinkedHashSet<>();
```

Comment:
In order to have full line selections for statement detection, we need to do the same text range extension that is done in extend word seleciton handlers.